### PR TITLE
UpdateUsers: allow partial update

### PR DESCRIPTION
**Pull request by Bogdan-D**
# Submit a pull request

Example

```go

// add a new field
update := PartialUserUpdate{
		ID: user.ID,
		Set: map[string]interface{}{
			"field": true,
		},
	}
user, err := c.PartialUpdateUser(update)

// remove a new field
update := PartialUserUpdate{
		ID: user.ID,
		Unset: []string{"field"},
	}
user, err := c.PartialUpdateUser(update)

// change nested field
update := PartialUserUpdate{
		ID: user.ID,
		Set: map[string]interface{}{
			"field.nested": "data",
		},
	}
user, err := c.PartialUpdateUser(update)


// replace object field
update := PartialUserUpdate{
		ID: user.ID,
		Set: map[string]interface{}{
			"field": map[string]interface{}{
                            "subfield": "data"
                        },
		},
	}
user, err := c.PartialUpdateUser(update)

...

```

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: user_test.go [hunk 1]
```diff
@@ -59,9 +59,43 @@ func TestClient_UpdateUsers(t *testing.T) {
...
+	t.Run("update users", func(t *testing.T) {
+		resp, err := c.UpdateUsers(user)
+		mustNoError(t, err, "update users")
+
+		assert.Contains(t, resp, user.ID)
+		assert.NotEmpty(t, resp[user.ID].CreatedAt)
+		assert.NotEmpty(t, resp[user.ID].UpdatedAt)
+	})
+
+	t.Run("partial update", func(t *testing.T) {
```
Comment (pascaldekloe) [inline user_test.go]: Run is not suitable for stacking tests. That is, this run depends on the previous.

File: user_test.go [hunk 2]
```diff
@@ -59,9 +59,43 @@ func TestClient_UpdateUsers(t *testing.T) {
...
+		extra := map[string]interface{}{
+			"test":   true,
+			"random": randomString(12),
+		}
+
+		resp, err := c.UpdateUsers(&User{ID: user.ID, ExtraData: extra})
+		mustNoError(t, err, "update users")
+
+		assert.Contains(t, resp, user.ID)
```
Comment (pascaldekloe) [inline user_test.go]: What does that even mean?

File: user_test.go [hunk 3]
```diff
@@ -59,9 +59,43 @@ func TestClient_UpdateUsers(t *testing.T) {
...
+		assert.Contains(t, resp[user.ID].ExtraData, "test", "extra data contains", resp[user.ID].ExtraData)
```
Comment (pascaldekloe) [inline user_test.go]: `new extra data field "test" not present in update response`

File: user_test.go [hunk 4]
```diff
@@ -59,9 +59,43 @@ func TestClient_UpdateUsers(t *testing.T) {
...
+	})
+
+	t.Run("partial update", func(t *testing.T) {
+		extra := map[string]interface{}{
+			"test":   true,
+			"random": randomString(12),
+		}
+
+		resp, err := c.UpdateUsers(&User{ID: user.ID, ExtraData: extra})
+		mustNoError(t, err, "update users")
```
Comment (pascaldekloe) [inline user_test.go]: "user update, with extra data insert"

File: user_test.go [hunk 5]
```diff
@@ -59,9 +59,43 @@ func TestClient_UpdateUsers(t *testing.T) {
...
+
+		assert.Contains(t, resp, user.ID)
+		assert.Contains(t, resp[user.ID].ExtraData, "test", "extra data contains", resp[user.ID].ExtraData)
+		assert.Contains(t, resp[user.ID].ExtraData, "random", "extra data contains", resp[user.ID].ExtraData)
+		assert.Equal(t, extra["test"], resp[user.ID].ExtraData["test"], "extra data equal", resp[user.ID].ExtraData)
```
Comment (pascaldekloe) [inline user_test.go]: `new extra data field "test" value changed in update response`

File: user.go [hunk 6]
```diff
@@ -254,3 +259,34 @@ func (c *Client) UpdateUsers(users ...*User) (map[string]*User, error) {
...
 	return resp.Users, err
 }
+
+type PartialUserUpdate struct {
+	ID    string                 `json:"id"`
+	Set   map[string]interface{} `json:"set,omitempty"`
+	Unset []string               `json:"unset,omitempty"`
+}
+
+func (c *Client) PartialUpdateUser(update PartialUserUpdate) (*User, error) {
```
Comment (tbarbugli) [inline user.go]: docs

File: user.go [hunk 7]
```diff
@@ -254,3 +259,34 @@ func (c *Client) UpdateUsers(users ...*User) (map[string]*User, error) {
 
 	return resp.Users, err
 }
+
+type PartialUserUpdate struct {
```
Comment (tbarbugli) [inline user.go]: docs

File: user.go [hunk 8]
```diff
@@ -254,3 +259,34 @@ func (c *Client) UpdateUsers(users ...*User) (map[string]*User, error) {
...
+	res, err := c.PartialUpdateUsers([]PartialUserUpdate{update})
+	if err != nil {
+		return nil, err
+	}
+
+	if user, ok := res[update.ID]; ok {
+		return user, nil
+	}
+
+	return nil, fmt.Errorf("response error: no user with such ID in response: %s", update.ID)
```
[the remark this answers is not in the document]
Comment (Bogdan-D) [reply user.go]: this one is about undefined behaviour, when response body doesn't contain required user to avoid panic

**Pull request merged**